Consul/Client_GetRequests.cs

```diff
@@ -79,8 +79,18 @@ public async Task<QueryResult<TOut>> Execute(CancellationToken ct)
             result.StatusCode = response.StatusCode;
             ResponseStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
 
-            if (response.StatusCode != HttpStatusCode.NotFound && !response.IsSuccessStatusCode)
+            var isAgentServiceName = Endpoint.StartsWith("v1/agent/health/service/name/", StringComparison.OrdinalIgnoreCase);
```

Comment:
I know this checks are fast, but I think it would be nice to do the string comparison only if necessary (e.g. use Lazy<T>, or lambda).

Reply:
I've tried something, but I am skeptical if there would be any performance gains from this, since the following line will be executed every time. I'd argue that there is even a regression now because we are making the function call twice, whereas before we were caching the results.

```cs
if (response.StatusCode != HttpStatusCode.NotFound && !response.IsSuccessStatusCode && !isSpecialStatusCode())
```

Reply:
Should lambdas be in PascalCase or camelCase?

Reply:
Looks like we use camelCase for lambdas in https://github.com/G-Research/consuldotnet/blob/master/Consul.Test/TimeoutUtils.cs#L36